src/clusterfuzz/_internal/bot/tasks/utasks/analyze_task.py

```diff
@@ -269,28 +267,37 @@ def execute_task(testcase_id, job_type):
   metadata.timestamp = datetime.datetime.utcnow()
   metadata.put()
 
+  initialize_testcase_for_main(testcase, job_type)
+
+  testcase_download_url = setup.get_signed_testcase_download_url(testcase)
+  return {
+      'metadata': metadata,
+      'testcase': testcase,
+      'uworker_env': uworker_env,
+      'testcase_download_url': testcase_download_url
+  }
+
+
+def utask_main(testcase, testcase_download_url, job_type, metadata):
+  """Executes the untrusted part of analyze_task."""
   prepare_env_for_main(metadata)
 
   is_lsan_enabled = environment.get_value('LSAN')
   if is_lsan_enabled:
     # Creates empty local blacklist so all leaks will be visible to uploader.
     leak_blacklist.create_empty_local_blacklist()
 
-  testcase_file_path = setup_testcase_and_build(testcase, metadata, job_type,
-                                                testcase_id)
+  testcase_file_path, output = setup_testcase_and_build(
```

Comment:
(nit, probably better addressed in a different CL): would a better variable name here be `results` (and perhaps `UworkerOutput`s should be `UworkerResult`s?  A `result` sounds like something that could contain output + error message, whereas `output` just sounds like it would be "something returned by a successful call", so I did a slight double-take trying to figure out why we'd return `output` when there's no `testcase_file_path`